backend/inspirehep/records/api/mixins.py

```diff
@@ -326,11 +328,24 @@ def get_newest_linked_conferences_uuid(self):
 
 
 class InstitutionPapersMixin:
+    institution_fields = [
+        "authors.affiliations.record",
+        "thesis_info.institutions.record",
+        "record_affiliations.record",
+    ]
+
     def clean_institution_literature_relations(self):
         InstitutionLiterature.query.filter_by(literature_uuid=self.id).delete()
 
     def create_institution_relations(self):
-        institutions = self.get_linked_records_from_field("authors.affiliations.record")
+        institutions = list(
+            chain.from_iterable(
+                [
+                    self.get_linked_records_from_field(field)
+                    for field in self.institution_fields
+                ]
+            )
```

Comment:
The list is not needed here:
```suggestion
            chain.from_iterable(
                self.get_linked_records_from_field(field)
                for field in self.institution_fields
            )
```